sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -390,13 +393,21 @@ class ProjectOverviewView extends VerticalLayout{
         }
         else if (sampleCount.passingSamples == sampleCount.totalSampleCount) {
             return State.COMPLETED
-        }
-        else if (sampleCount.passingSamples < sampleCount.totalSampleCount) {
+        } else if (sampleCount.passingSamples < sampleCount.totalSampleCount) {
             return State.IN_PROGRESS
-        }
-        else {
+        } else {
             //unexpected!!
             throw new IllegalStateException("status count $sampleCount.passingSamples must not be greater total count $sampleCount.totalSampleCount")
         }
     }
+
+    private void setupFilters(ListDataProvider<ProjectSummary> projectSummaryListDataProvider) {
```

Comment:
The method signature here can be simplified. As the class variable `projectGrid` is used here, it can be queried for its data provider which can be casted `as ListDataProvider<ProjectSummary>`. When working with the class variable no additional arguments are needed here.

Another possibility would be to move this method to the GridUtils and provide it with a `Grid<T>` and a `Collection<String>` containing the grid to be set up and the column identifiers.

Reply:
I would prefer the second option as this does not introduce as many side effects.

Reply:
Nice suggestion!